File: frontend/src/services/transactionService.js
import api from './api';

const transactionService = {
  deposit: async (transactionData) => {
    const response = await api.post('/transactions/deposit', transactionData);
    return response.data;
  },

  withdraw: async (transactionData) => {
    const response = await api.post('/transactions/withdraw', transactionData);
    return response.data;
  },

  getTransactionHistory: async (accountNumber, page = 0, size = 10) => {
    const response = await api.get(`/transactions/${accountNumber}/history`, {
      params: { page, size },
    });
    return response.data;
  },
};

export default transactionService;

File: backend/src/main/java/com/bank/transaction/service/TransactionService.java
package com.bank.transaction.service;

import com.bank.transaction.dto.TransactionRequest;
import com.bank.transaction.dto.TransactionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface TransactionService {
    TransactionResponse deposit(TransactionRequest request);
    TransactionResponse withdraw(TransactionRequest request);
    Page<TransactionResponse> getTransactionHistory(String accountNumber, Pageable pageable);
}
